### PR TITLE
python-libgpiod: add python3 to nativeBuildInputs; add to PYTHONPATH; resolves #166049

### LINKED ISSUE #166049: Cross compiling Python `libgpiod` for Raspberry Pi fails
### Describe the bug
Cross compiling Python `libgpiod` for Raspberry Pi fails with
```
checking for a Python interpreter with version >= 3.0... none
```

### Steps To Reproduce
```bash
# build from current master (Mon Mar 28 08:55:41 AM UTC 2022)
nix-build https://github.com/NixOS/nixpkgs/archive/e801b16.tar.gz -A pkgsCross.raspberryPi.python3Packages.libgpiod
```

### Expected behavior
A successful build of the package with Python support.

### Screenshots
image

### Additional context
```bash
# same problem with 21.11
nix-build https://github.com/NixOS/nixpkgs/archive/nixos-21.11.tar.gz -A pkgsCross.raspberryPi.python3Packages.libgpiod
```

### Notify maintainers
@expipiplus1
@ryantm
@jonringer

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.106, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(marble): `"nixos-unstable-22.05pre356180.d5f23787297"`
 - channels(root): `"nixos-21.11.336646.d2caa937753"`
 - channels(work): `"nixos-unstable-21.11pre328987.c935f5e0add"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by cyber-murmel**
###### Description of changes
Add `python3` optionally to `nativeBuildInputs` to fix cross compilation for armv6l and armv7l (#166049).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mindavi — CHANGES_REQUESTED** [review]
Based on comments in issue this needs some work still

**Comment by cyber-murmel** [comment]
The problem with the import was, that the PYTHONPATH wasn't modified. import works now.

**Comment by FRidh** [comment]
Can you set `strictDeps = true;`

File: pkgs/development/libraries/libgpiod/default.nix [hunk 1]
```diff
@@ -22,14 +22,19 @@ stdenv.mkDerivation rec {
...
-  ];
+  ] ++ lib.optionals enablePython [ python3 ];
 
   configureFlags = [
     "--enable-tools=${if enable-tools then "yes" else "no"}"
     "--enable-bindings-cxx"
     "--prefix=${placeholder "out"}"
   ] ++ lib.optional enablePython "--enable-bindings-python";
 
+  postInstall = lib.optional enablePython ''
```
Comment (FRidh) [inline pkgs/development/libraries/libgpiod/default.nix:33]: This should actually be done in the hook.

Comment (cyber-murmel) [reply pkgs/development/libraries/libgpiod/default.nix]: I updated the PR. `python3Packages.gpiod` now is more like `python3Packages.sigrok`

**Review by SuperSandro2000 — COMMENTED** [review]
Why are we not using a flag `withPython` ? This could make things a lot easier.

**Comment by cyber-murmel** [comment]
@SuperSandro2000 So would it be preferred to add the stuff (that's missing for successful cross compilation) to the normal libgpiod package with behind `lib.optionals withPython`?
I don't know yet what's best practice in this regard.

**Comment by SuperSandro2000** [comment]
I believe so. That should cut down on code and duplication which is almost always a good thing.

File: pkgs/development/libraries/libgpiod/default.nix [hunk 2]
```diff
@@ -3,7 +3,7 @@
 , enablePython ? false, python3, ncurses }:
 
 stdenv.mkDerivation rec {
-  pname = "libgpiod";
+  pname = lib.optionals enablePython "${python3.libPrefix}-" + "libgpiod";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgpiod/default.nix:6]: I'd rather move this to pythonPackages if possible then we don't need this.

**Pull request closed without merging**